pages/agent/v3/elastic_ci.md.erb

```diff
@@ -0,0 +1,275 @@
+
+These files are encrypted using [Amazon's KMS Service](https://aws.amazon.com/kms/). See the [Security](#security) section for more details.
+
+Here's an example that shows how to generate a private SSH key, and upload it with KMS encryption to an S3 bucket:
+
+```bash
+# generate a deploy key for your project
+ssh-keygen -t rsa -b 4096 -f id_rsa_buildkite
+pbcopy < id_rsa_buildkite.pub # paste this into your github deploy key
+
+aws s3 cp --acl private --sse aws:kms id_rsa_buildkite "s3://${SecretsBucket}/private_ssh_key"
+```
+
+If you want to set secrets that your build can access, create a file that sets environment variables and upload it:
+
+```bash
+echo "export MY_ENV_VAR=something secret" > myenv
+aws s3 cp --acl private --sse aws:kms myenv "s3://${SecretsBucket}/env"
+rm myenv
+```
+<div class="Docs__note">
+Currently, you must use the default KMS key for S3. follow [#235](https://github.com/buildkite/elastic-ci-stack-for-aws/issues/235) for progress on using specific KMS keys.
+</div>
+
+If you want to store your secrets unencrypted, you can disable it entirely with `BUILDKITE_USE_KMS=false`.
+
+## What’s on each machine?
+
+* [Amazon Linux 2 LTS](https://aws.amazon.com/amazon-linux-2/)
+* [Buildkite Agent v3.27.0](https://buildkite.com/docs/agent)
+* [Docker](https://www.docker.com) - 19.03.13 (Linux) and 19.03.12 (Windows)
+* [Docker Compose](https://docs.docker.com/compose/) - 1.27.4 (Linux) and 1.27.2 (Windows)
+* [AWS CLI](https://aws.amazon.com/cli/) - useful for performing any ops-related tasks
+<!-- vale off -->
+* [jq](https://stedolan.github.io/jq/) - useful for manipulating JSON responses from CLI tools such as AWS CLI or the Buildkite API
+<!-- vale on -->
+
+## What type of builds does this support?
+
+This stack is designed to run your builds in a share-nothing pattern similar to the [12 factor application principals](http://12factor.net):
+
+* Each project should encapsulate its dependencies through Docker and Docker Compose.
+* Build pipeline steps should assume no state on the machine (and instead rely on [build meta-data](/docs/guides/build-meta-data), [build artifacts](/docs/guides/artifacts) or S3).
+* Secrets are configured via environment variables exposed using the S3 secrets bucket.
+
+By following these conventions you get a scalable, repeatable, and source-controlled CI environment that any team within your organization can use.
+
+## Multiple instances of the stack
+
+If you need different instances sizes and scaling characteristics for different pipelines, you can create multiple stacks. Each can run on a different [Agent queue](/docs/agent/queues), with its own configuration, or even in a different AWS account.
+
+Examples:
+
+* A `docker-builders` stack that provides always-on workers with hot Docker caches (see [Optimizing for slow Docker builds](#optimizing-for-slow-docker-builds))
+* A `pipeline-uploaders` stack with tiny, always-on instances for lightning fast `buildkite-agent pipeline upload` jobs.
```

Comment:
I wonder if @yob could find us links to examples for these other two?

Reply:
Or @jayco ?

Reply:
I'm not sure if we have any public documentation on these kinds of queues and agent patterns. I suspect we may not.

@jayco is definitely the one to ask! @toolmantim might also know.

Reply:
Hmm, I don't think we actually have any examples of this. 

I normally point to optimizing-for-slow-docker-builds that you have already linked to.